### PR TITLE
Webhook response customisation

### PR DESCRIPTION
**Pull request by gdebrauwer**
This PR is inspired by the PR #44.

It allows you to customize the webhook response. This makes it possible to use this package in situations where a special webhook response is needed.

### CODE REVIEW

File: src/WebhookResponse/WebhookResponse.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Spatie\WebhookClient\WebhookResponse;
+
+use Illuminate\Http\Request;
+use Spatie\WebhookClient\WebhookConfig;
+
+interface WebhookResponse
+{
+    public function respondToValidWebhookRequest(Request $request, WebhookConfig $config);
```
Comment (freekmurze) [inline src/WebhookResponse/WebhookResponse.php:10]: Let's add `Symfony\Component\HttpFoundation\Response` as the return type. Also add this return type to all approriate functions that return the response.

**Comment by freekmurze** [comment]
Thank you!

**Comment by gdebrauwer** [comment]
@freekmurze Should this not have been tagged as a new major version? 🤔It's not an issue for me, but maybe for others that update their composer packages without knowing about this new, required config-field?

**Comment by freekmurze** [comment]
The config in your app will be merged with the one in the package

https://github.com/spatie/laravel-webhook-client/blob/ac101820ceee6e23e953af72463f3e28863e157e/src/WebhookClientServiceProvider.php#L59

So this shouldn't be a breaking change for people that already have the package installed.

**Comment by freekmurze** [comment]
Does this fix it for you? https://github.com/spatie/laravel-webhook-client/commit/d21404f29e1c058097b9c4c17fc20c49cc9e851f

(thanks for bringing this problem to my attention)

**Comment by gdebrauwer** [comment]
No, it's fixed if I do the following:

```php
use Spatie\WebhookClient\WebhookResponse\DefaultRespondsTo;

$webhookResponseClass = $properties['webhook_response'] ?? DefaultRespondsTo::class;
if (! is_subclass_of($webhookResponseClass, RespondsToWebhook::class)) {
    throw InvalidConfig::invalidWebhookResponse($webhookResponseClass);
}
$this->webhookResponse = app($webhookResponseClass);
```

**Comment by freekmurze** [comment]
Could you send a quick PR for that?

**Comment by gdebrauwer** [comment]
Yes 👍

**Pull request merged**